src/main/java/org/dependencytrack/persistence/VulnerabilityQueryManager.java

```diff
@@ -423,6 +423,7 @@ public List<Vulnerability> getVulnerabilities(Project project, boolean includeSu
             );
             for (final Vulnerability componentVuln: componentVulns) {
                 componentVuln.setComponents(Collections.singletonList(pm.detachCopy(component)));
+                componentVuln.setAliases(getVulnerabilityAliases(componentVuln));
```

Comment:
@sahibamittal The aliases need to be detached, otherwise we'll be leaking resources.

```suggestion
                componentVuln.setAliases(new ArrayList<>(pm.detachCopyAll(getVulnerabilityAliases(componentVuln))));
```